app/src/main/java/com/example/mvvmcleanarchitecture/core/di/AppModule.kt

```diff
@@ -0,0 +1,28 @@
+package com.example.mvvmcleanarchitecture.core.di
+
+import android.content.Context
+import androidx.recyclerview.widget.DividerItemDecoration
+import androidx.recyclerview.widget.GridLayoutManager
+import androidx.recyclerview.widget.LinearLayoutManager
+import org.koin.android.ext.koin.androidContext
+import org.koin.dsl.module
+
+val appModule = module {
+    factory { provideLinearLayoutManager(androidContext()) }
```

Comment:
Czy dobrym pomysłem byłoby wydzielenie tworzenia obiektów do metod ?

Reply:
W zasadzie nie widzę przeciwwskazań, żeby tego nie robić. Ja osobiście wolę od razu deklarować w ciele `factory` / `single`, ale jeśli bardziej Cię przekonuje wydzielenie tego do oddzielnej metody to jest to jak najbardziej ok. Ja coś takiego robię tylko w przypadku, gdy tworzę bibliotekę / SDK i muszę dostarczyć jakiś element modułu przy inicjalizacji takiej libki.